schemas/app-sre/saas-file-2.yml

```diff
@@ -10,111 +18,198 @@ properties:
+      such as for `oc process` or `helm template`.
+
   allowedSecretParameterPaths:
     type: array
     items:
       type: string
+    description: |
+      A list of allowed Vault paths for secret parameters.
+
   secretParameters:
     type: array
-    description: saas file level parameters from vault secrets
+    description: |
+      A list of secret template parameters, injected from Vault, that are applied to all resource templates
+      in this SaaS file.
     items:
       "$ref": "/app-sre/vault-secret-parameter-1.yml"
+
   validateTargetsInApp:
-    description: indicates a desire to validate that all targets belong to the same app as the saas file itself
     type: boolean
+    description: |
+      If true, validates that all deployment targets belong to the same
+      app as the SaaS file.
+
   validatePlannedData:
     type: boolean
-    description: enable skipping planned data validation, mostly for automated bootstrap pusrposes
+    description: |
+      If false, skips validation of planned data, such as dependent
+      resources (i.e. a Secret mounted by a Deployment). Useful for automated bootstrap purposes. 
+
   resourceTemplates:
     type: array
+    description: |
+      A list of resource templates to deploy. Each template specifies
+      its source, parameters, and deployment targets.
     items:
       type: object
       additionalProperties: false
       properties:
         name:
           type: string
+          description: |
+            A descriptive name for the deployed resources.
         url:
           type: string
           format: uri
+          description: |
+            The git repository URL containing the resource template.
         path:
           type: string
           pattern: '^\/.*'
+          description: |
+            The path to the file containing the resource template in the repository.
         provider:
           type: string
           enum:
           - openshift-template
           - directory
           - helm
+          description: |
+            The type of resource template: OpenShift template, directory of
+            manifests, or Helm chart.
         hash_length:
           type: integer
+          description: |
+            The length of the commit hash to use for image tags.
         parameters:
           type: object
+          description: |
+            Parameters to use when deploying this resource template.
         secretParameters:
           type: array
-          description: resource template level parameters from vault secrets
+          description: |
+            A list of secret template parameters, injected from Vault, that are applied to
+            the associated resource template targets.
           items:
             "$ref": "/app-sre/vault-secret-parameter-1.yml"
         targets:
           type: array
+          description: |
+            A list of deployment targets for this resource template.
```

Comment:
Maybe mention that in most cases targets can be set inline instead of externally to the saas file